daffodil-runtime1/src/main/scala/org/apache/daffodil/processors/DataProcessor.scala

```diff
@@ -129,10 +131,10 @@ class DataProcessor private (
   // The values these will have (since this is a base class) are the correct default values that we want
   // back when the object is re-initialized.
   //
-  protected val areDebugging : Boolean,
-  protected val optDebugger : Option[Debugger],
+  private val externalVars: Queue[Binding],
```

Comment:
I think maybe `externalVars` can go away completely? The only way to set external variables should be with `withExternalVariables`, which I think should already create a copy of the variable map and mutate it with the new bindings?

I *think* externalVars was only needed because the compiler/processor factory could have variables set on them, so they needed to be carried down this way to the DataProcessor?

So maybe the `SchemaSetRuntimeData` just passes in the `originalVariables` Variable map when it creates the DataProcosser, and that is copied/mutated as needed?